control.cc

```diff
@@ -661,7 +661,11 @@ int relatedclass::parse(ifstream &inf, ostream &outf, ostream &outd)
 				outd << "START " << tname << endl;
 				saved_name = tname;
 
-				if(tname.length() == 0) { // medm ignores blank names. edm does not.
+				if(tname.length() != 0) { // medm ignores blank names. edm does not.
+					adl_pos = tname.find(adl,0);
+					if(adl_pos != -1){
+						tname.replace(adl_pos,4,echar);
+					}
```

Comment:
I won't pretend to understand what this code does (or did).  Is this just stripping any '.adl' suffix?  All the other changes look straightforward.

Reply:
It is replacing an ".adl" suffix with ".edl" - which sounds very reasonable for adl2edl.